Autoplay to next video from full screen causes it to exit to inline

Phinny01 opened the pull request:
#### b805ee18766623b1bf25c745294297443923d807
<pre>
Autoplay to next video from full screen causes it to exit to inline
<a href="https://bugs.webkit.org/show_bug.cgi?id=304056">https://bugs.webkit.org/show_bug.cgi?id=304056</a>
<a href="https://rdar.apple.com/162684936">rdar://162684936</a>

Reviewed by Brent Fulgham.

IMDb&apos;s fullscreen video player exits fullscreen prematurely when autoplaying to the next video on
iPad. This is caused by JWPlayer detecting iOS using `navigator.maxTouchPoints &gt; 1 &amp;&amp;
navigator.platform === &quot;MacIntel&quot;` and applying iOS-specific code that exits fullscreen when
videos end.

Adding imdb.com to `needsZeroMaxTouchPointsQuirk` resolves this issue by returning 0 for
maxTouchPoints, preventing incorrect iOS detection in desktop mode on iPad.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsZeroMaxTouchPointsQuirk const):
(WebCore::handleIMDBQuirks):
* Source/WebCore/page/QuirksData.h:

Canonical link: <a href="https://commits.webkit.org/304393@main">https://commits.webkit.org/304393@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a9ed366bd589a48a84a66d0d70ebdf298cdee24e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/135438 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/7823 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/46719 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/143131 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/87137 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/aefcdd95-d7b8-46cd-baa1-4e4ce952fc46) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/137307 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/8453 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/7663 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/103502 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/87137 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/138384 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/6059 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/121388 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/84368 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/0194fbcc-821a-4d72-a70a-3e8303253f62) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/5839 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/3448 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/3742 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/115040 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/39575 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/145886 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/7499 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/40145 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/111872 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/7540 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/6305 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/112243 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28483 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/5692 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/117690 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/61397 "The change is no longer eligible for processing.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/7553 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/35810 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/7308 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/71101 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/7519 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/7409 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -3065,6 +3035,7 @@ static void handleIMDBQuirks(QuirksData& quirksData, const URL& /* quirksURL */,
 
     // imdb.com: rdar://137991466
     quirksData.enableQuirk(QuirksData::SiteSpecificQuirk::NeedsChromeMediaControlsPseudoElementQuirk);
+    quirksData.enableQuirk(QuirksData::SiteSpecificQuirk::NeedsZeroMaxTouchPointsQuirk);
```
This is fine, but a more stylish approach is:

```
    quirksData.enableQuirks({
        QuirksData::SiteSpecificQuirk::NeedsChromeMediaControlsPseudoElementQuirk,
        QuirksData::SiteSpecificQuirk::NeedsZeroMaxTouchPointsQuirk
    });

```

brentfulgham replied:
Please list `// imdb.com: rdar://137991466` for the `NeedsChromeMediaControlsPseudoElementQuirk` and `rdar://162684936` for the `NeedsZeroMaxTouchPointsQuirk`.

Phinny01 replied:
I have listed them now

brentfulgham reviewed (approved):
Looks great! Thank you for figuring this out. I had a really minor style suggestion, but this is good as-is.

Commit:
Autoplay to next video from full screen causes it to exit to inline
https://bugs.webkit.org/show_bug.cgi?id=304056
rdar://162684936

Reviewed by Brent Fulgham.

IMDb's fullscreen video player exits fullscreen prematurely when autoplaying to the next video on
iPad. This is caused by JWPlayer detecting iOS using `navigator.maxTouchPoints > 1 &&
navigator.platform === "MacIntel"` and applying iOS-specific code that exits fullscreen when
videos end.

Adding imdb.com to `needsZeroMaxTouchPointsQuirk` resolves this issue by returning 0 for
maxTouchPoints, preventing incorrect iOS detection in desktop mode on iPad.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsZeroMaxTouchPointsQuirk const):
(WebCore::handleIMDBQuirks):
* Source/WebCore/page/QuirksData.h:

Canonical link: https://commits.webkit.org/304393@main
